src/main/webapp/WEB-INF/tags/shared/commentsPanel.tag

```diff
@@ -55,7 +55,7 @@
                         <shared:commentRow comment="${commentRow}" commentIndex="${commentIndex}" />
                     </c:forEach>
                     <c:if test="${forRecordsPage}">
-                        <c:if test="${empty commentsForStudentsTable.rows}">
+                        <c:if test="${empty commentsForStudentsTable.rows && commentsForStudentsTable.isRepresentingSelf}">
```

Comment:
If you do rename `getIsRepresentingSelf` to `isRepresentingSelf`, you will probably have to change this to `commentsForStudentsTable.representingSelf`